awx/ui_next/src/components/Schedule/shared/FrequencyDetailSubform.jsx

```diff
@@ -0,0 +1,445 @@
+              aria-label={i18n._(t`Wednesday`)}
+              id="days-of-week-wed"
+              name="daysOfWeek"
+            />
+            <Checkbox
+              label={i18n._(t`Thu`)}
+              isChecked={daysOfWeek.value.includes('TH')}
+              onChange={checked => {
+                updateDaysOfWeek('TH', checked);
+              }}
+              aria-label={i18n._(t`Thursday`)}
+              id="days-of-week-thu"
+              name="daysOfWeek"
+            />
+            <Checkbox
+              label={i18n._(t`Fri`)}
+              isChecked={daysOfWeek.value.includes('FR')}
+              onChange={checked => {
+                updateDaysOfWeek('FR', checked);
+              }}
+              aria-label={i18n._(t`Friday`)}
+              id="days-of-week-fri"
+              name="daysOfWeek"
+            />
+            <Checkbox
+              label={i18n._(t`Sat`)}
+              isChecked={daysOfWeek.value.includes('SA')}
+              onChange={checked => {
+                updateDaysOfWeek('SA', checked);
+              }}
+              aria-label={i18n._(t`Saturday`)}
+              id="days-of-week-sat"
+              name="daysOfWeek"
+            />
+          </div>
+        </FormGroup>
+      )}
+      {(frequency?.value === 'month' || frequency?.value === 'year') &&
+        !isNaN(new Date(startDateTime.value)) && (
+          <FormGroup
+            name="runOn"
+            fieldId="schedule-run-on"
+            helperTextInvalid={runOnMeta.error}
+            isRequired
+            isValid={!runOnMeta.touched || !runOnMeta.error}
+            label={i18n._(t`Run on`)}
+          >
+            <Radio
+              id="run-on-number"
+              name="runOn"
+              label={generateRunOnNumberLabel()}
+              value="number"
+              isChecked={runOn.value === 'number'}
+              onChange={(value, event) => {
+                event.target.value = 'number';
+                runOn.onChange(event);
+              }}
+            />
+            <Radio
+              id="run-on-day"
+              name="runOn"
+              label={generateRunOnDayLabel()}
+              value="day"
+              isChecked={runOn.value === 'day'}
+              onChange={(value, event) => {
+                event.target.value = 'day';
+                runOn.onChange(event);
+              }}
+            />
+            {new Date(startDateTime.value).getDate() >
+              getDaysInMonth(startDateTime.value) - 7 && (
+              <Radio
+                id="run-on-last-day"
+                name="runOn"
+                label={generateRunOnLastDayLabel()}
+                value="lastDay"
+                isChecked={runOn.value === 'lastDay'}
+                onChange={(value, event) => {
+                  event.target.value = 'lastDay';
+                  runOn.onChange(event);
+                }}
+              />
+            )}
+          </FormGroup>
+        )}
```

Comment:
For monthly frequency and possibly others, I think parts of this should be actual text input fields(in addition to the radio buttons, e.g:

![Screenshot from 2020-03-25 13-56-54](https://user-images.githubusercontent.com/9753817/77569536-bb096800-6ea0-11ea-9735-b27d1694fc83.png)

Reply:
This might actually have to do with:

>  If you want to run a monthly schedule on the 15th of each month then you have to specify a start date/time on the 15th of the month. If you want to run a yearly schedule on Jan 1 then you have to specify a start date/time of Jan 1.

If so nevermind

Reply:
I like Jake's solution above, as well. maybe with less intimidatingly-wide fields

Reply:
This also applies to the yearly solution as well.

Reply:
discussed this w/ @mabashian - we're all definitely on the same page here. The key bit of info is that the datepicker component is currently WIP patternfly side but we'll be integrating it soon at a later date and outside of this pr.

Reply:
https://github.com/ansible/awx/issues/6449

Reply:
Yea, I think i'm going to move back towards the UX of the old app.  I do think that you can achieve the same rules with both UX's but the old app is more flexible and I'd like to potentially incorporate https://github.com/ansible/awx/issues/6402 into the implementation.